cli/crates/server/src/udf_builder.rs

```diff
@@ -325,23 +351,41 @@ pub async fn install_wrangler(environment: &Environment, tracing: bool) -> Resul
     .await?
     .map_err(ServerError::Lock)?;
 
+    if let Some(installed_wrangler_version) =
+        installed_wrangler_version(&environment.wrangler_installation_path, tracing).await
+    {
+        info!("Installed wrangler version: {installed_wrangler_version}");
+        if installed_wrangler_version == WRANGLER_VERSION {
+            info!("wrangler of the desired version already installed, skipping…");
+            return Ok(());
+        }
+    }
+
+    let wrangler_installation_path_str = environment.wrangler_installation_path.to_str().expect("must be valid");
+
     info!("Installing wrangler…");
     tokio::fs::create_dir_all(&environment.wrangler_installation_path)
         .await
         .map_err(|_| ServerError::CreateDir(environment.wrangler_installation_path.clone()))?;
     // Install wrangler once and for all.
     run_command(
         JavaScriptPackageManager::Npm,
-        &["add", "--save-dev", "wrangler@2"],
-        environment.wrangler_installation_path.to_str().expect("must be valid"),
+        &[
+            "add",
+            "--save-dev",
+            &format!("wrangler@{WRANGLER_VERSION}"),
+            "--prefix",
+            wrangler_installation_path_str,
+        ],
+        wrangler_installation_path_str,
         tracing,
         &[],
     )
     .await?;
     run_command(
         JavaScriptPackageManager::Npm,
-        &["install"],
-        environment.wrangler_installation_path.to_str().expect("must be valid"),
+        &["install", "--prefix", wrangler_installation_path_str],
+        wrangler_installation_path_str,
```

Comment:
I was under the impression `npm add` was an alias for `npm install` - is this second call actually doing anything?

Reply:
@obmarg silly me, of course.